MaxText/train.py

```diff
@@ -84,13 +85,18 @@ def get_first_step(state):
     return int(state.step)
 
 
-def load_next_batch(train_iter, example_batch, config):
+def load_next_batch(train_iter, example_batch, config, mesh):
   """Loads the next batch. Can keep reusing the same batch for performance reasons """
 
   if config.reuse_example_batch and example_batch is not None:
     return example_batch
   else:
-    return train_iter()
+    if config.dataset_type == 'c4':
+      return train_iter()
+    elif config.dataset_type == 'c4-array_record':
+      return get_next_batch_sharded(train_iter, mesh)
```

Comment:
I feel like this interface should be unified -- this code shouldn't know about dataset_type

Reply:
revised as suggested.